lib/membrane_h264_ffmpeg/parser.ex

```diff
@@ -132,19 +130,21 @@ defmodule Membrane.H264.FFmpeg.Parser do
         state.first_frame_prefix <> payload
       end
 
-    with {:ok, sizes} <- Native.parse(payload, state.parser_ref) do
-      {bufs, state} = parse_access_units(payload, sizes, metadata, state)
+    case Native.parse(payload, state.parser_ref) do
+      {:ok, sizes} ->
+        {bufs, state} = parse_access_units(payload, sizes, metadata, state)
 
-      caps =
-        if ctx.pads.output.caps == nil and bufs != [] do
-          [caps: {:output, mk_caps(state)}]
-        else
-          []
-        end
+        caps =
+          if ctx.pads.output.caps == nil and bufs != [] do
+            [caps: {:output, mk_caps(state)}]
+          else
+            []
+          end
 
-      {{:ok, caps ++ [buffer: {:output, bufs}, redemand: :output]}, state}
-    else
-      {:error, reason} -> {{:error, reason}, state}
+        {{:ok, caps ++ [buffer: {:output, bufs}, redemand: :output]}, state}
+
+      {:error, reason} ->
+        {{:error, reason}, state}
```

Comment:
Was that credo suggestion? If so, you didn't use the config pushed here as it disables this check. We decided to stick to `with` as it explicitly separates "happy path" from error and is a bit better for refactors in future